### PR TITLE
cql3: coroutinize query_processor::for_each_cql_result

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
i just came across this two functions. and found it really difficult to read without being coroutinized. hence this change.

**Comment by tchaikov** [comment]
> CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/812/

failure tracked by #13594

**Comment by tchaikov** [comment]
@amnonh hi Amnon, could you help review this change?

File: cql3/query_processor.cc [hunk 1]
```diff
@@ -732,62 +732,30 @@ bool query_processor::has_more_results(::shared_ptr<cql3::internal_query_state>
...
-                        }
-                        ++it;
-                        return std::optional<stop_iteration>();
-                    });
-                });
-            });
-        });
-    });
+    do {
+        auto msg = co_await execute_paged_internal(state);
```
Comment (denesb) [inline cql3/query_processor.cc:752]: We either need the `msg->empty()` check here too, or we don't need it above.

Comment (avikivity) [reply cql3/query_processor.cc:752]: I think it's okay. It's a direct translation and coroutinization shouldn't try to perform extra optimizations.

As it happens I think the empty check is a bug. Suppose we receive an empty page but has_more_pages = true? But it should be fixed separately.

Comment (avikivity) [reply cql3/query_processor.cc:752]: There's also no need to have these two overloads. The only difference is returning future<stop_iteration> vs. non-future. The callers can be adjusted, or one can call the other.

Comment (tchaikov) [reply cql3/query_processor.cc:752]: I will fix both in a follow up pr with two different commits.

Comment (avikivity) [reply cql3/query_processor.cc:752]: It's better to avoid mixing fixes and refactoring.

**Review by denesb — APPROVED** [review]

File: cql3/query_processor.cc (hunk 1 above)
Comment (tchaikov) [reply cql3/query_processor.cc:752]: sure. will create a follow-up PR once this one is merged.

Comment (tchaikov) [reply cql3/query_processor.cc:752]: > sure. will create a follow-up PR once this one is merged.

#13675

**Pull request closed without merging**